python/pyspark/pandas/frame.py

```diff
@@ -8412,7 +8430,11 @@ def update(self, other: "DataFrame", join: str = "left", overwrite: bool = True)
             *HIDDEN_COLUMNS,
         )
         internal = self._internal.with_new_sdf(sdf, data_fields=data_fields)
-        self._update_internal_frame(internal, requires_same_anchor=False)
+        # Since Spark 3.4, df.update generates a new dataframe instead of operating
+        # in-place to follow pandas v1.4 behavior, see also SPARK-38946.
+        self._update_internal_frame(
+            internal, requires_same_anchor=False, anchor_force_disconnect=True
```

Comment:
We don't need to specify `requires_same_anchor` any more if `anchor_force_disconnect=True`?

Reply:
See https://github.com/apache/spark/pull/36353#issuecomment-1178956945 , after confirm from panda community, we only add `setitem` make copy. so this line had been removed